### DISCUSSION TITLE
OperationalError encountered twice. Never seen before on my setup- what's going on?

### DISCUSSION THREAD

**Discussion by wafflecrate** [inventree/InvenTree#5021] [General]
Hi everyone-

I'm currently running InvenTree 0.11.0. Full copy/pasted details:

# Version Information:
InvenTree-Version: 0.11.0
Django Version: 3.2.18
Database: postgresql
Debug-Mode: False
Deployed using Docker: False
Active plugins: [{'name': 'InvenTreeBarcode', 'slug': 'inventreebarcode', 'version': '2.0.0'}, {'name': 'InvenTreeCoreNotificationsPlugin', 'slug': 'inventreecorenotificationsplugin', 'version': '1.0.0'}]

I have it running on a DigitalOcean Droplet. I've encountered [an error that I've never seen before:](https://imgur.com/fImRXdh)

[Error 1 details](https://imgur.com/nDOH0Ow)
[Error 2 details](https://imgur.com/3odTK83)

Both errors occurred when my InvenTree system was NOT in use, which makes it all the more unusual. Once on June 7th, and another time on June 11th. As of now my InvenTree system is working without issue but since this is a live system running for a business I would like to figure out what is causing these OperationalErrors.

Any guesses as to what could be happening and what I can do to fix/prevent it from re-occurring?

**Comment by SchrodingersGat** [comment]
In both cases the database connection is failing due to a timeout error. Maybe the server is resource constrained? Or, there are not enough worker connections available and the database is temporarily overloaded?

The URL in your second example looks strange - is that an "expected" URL in your setup?

**Reply by matmair** [reply]
Seems like someone is probing the server from the URLs.
If it is reachable without extra security, I would strongly recommend enabling 2 factor and making sure your droplet password is long enough @wafflecrate

**Reply by matmair** [reply]
Ref: https://packetstormsecurity.com/files/162094/Ignition-2.5.1-Remote-Code-Execution.html

**Reply by SchrodingersGat** [reply]
If someone is hitting your login URL with a lot of requests, this would explain why your database is throwing timeout errors

**Comment by wafflecrate** [comment]
Thank you for the explanations. @SchrodingersGat @matmair. I've taken both of those extra steps to beef up my account security. Can these errors interfere with the functionality of my InvenTree database in any way, or is it just one of those inescapable security risks that I just need to deal with?

**Reply by matmair** [reply]
If the server is overloaded you will have a hard time accessing it. Depending on your needs it might be beneficial to restrict the IP ranges or countries that can access your instance in your nginx settings.
